src/cognito/auth.ts

```diff
@@ -0,0 +1,185 @@
+import { AuthenticationDetails, CognitoUser, CognitoUserPool } from 'amazon-cognito-identity-js';
+import { getEnv } from '../utils/common';
+import { isBeta } from '../utils/common';
+
+async function fetchData() {
+  try {
+    const response = await fetch(`${isBeta() ? '/beta' : ''}/apps/chrome/env.json`);
+    const jsonData = await response.json();
+    return jsonData;
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+function getSearchParams(url: string): { [key: string]: string } {
+  const searchString = new URL(url).search;
+  const searchParams = new URLSearchParams(searchString);
+  const searchParamsObject: { [key: string]: string } = {};
+  for (const [key, value] of searchParams) {
+    searchParamsObject[key] = value;
+  }
+  return searchParamsObject;
+}
+
+export async function getTokenWithAuthorizationCode() {
+  const code = getSearchParams(window.location.href).code;
+  const refreshToken = localStorage.getItem('REFRESH_TOKEN');
```

Comment:
Is the `REFRESH_TOKEN` set by the cognito or did we pick that specific key name? I would to have a slightly more unique local storage key if possible.